### ISSUE TITLE
Gloss translation popping up incorrectly

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#583]
Now this was funny: I've just created three new datasets (KSL, JSL, TSL). The description language Japanese was added for all three, as the researcher is Japanese. Then I selected these three datasets plus NGT, search for a gloss that only exists in NGT (search string ^PT), and now for the various ^PT glosses, the NGT dataset suddenly appears to have Japanese as a translation language in the list view! Haha! When I go to the detail view, it's not there; nor is it in Django when I look at the properties of the NGT dataset. I love languages, but this is getting out of hand! ;-)

**Comment by Woseseltops** [comment]
Let me see if I understand this correctly. I did the following things:
1. Selected the datasets NGT (description language: Dutch), BISINDO (description language: Indonesian), LSFB (description language: French) and JSL (description language: Japanese).
2. I searched for the coolest and best gloss in Signbank, which is part of the NGT dataset. This is what I see:

![image](https://user-images.githubusercontent.com/1909317/59910756-d7104280-9412-11e9-8eac-c75a765fdbbd.png)

Observations:
1. There is an 'Annotation ID gloss' column for each of the description languages in the datasets that I chose. 
2. All 'Annotation ID gloss' columns are filled, even if there is no info from the database (copied from the English one?).
3. There is a 'Lemma ID gloss' column for each of the description languages in the datasets that I chose. 4. All 'Lemma ID gloss' columns all emtpty except when there actually is info from the database.

Which of these 4 observations would you like to change? Observation 2?

**Comment by susanodd** [comment]
I tried this too.
Showing all of the columns for all of the languages of the selected datasets is what's supposed to happen, isn't it?
From Observation 2, that would be the default language showing up. It's often coded so that the default language is used if there isn't a value.

**Comment by ocrasborn** [comment]
Oh yes, the column should indeed be there if one of the selected datasets has that language (here, Japanese) as one of the description languages. But the field should not appear as if it is used for other datasets (here, NGT) that do not have that language as one of the description languages.

**Comment by ocrasborn** [comment]
In relation to the four observations by @Woseseltops : indeed the second one I don't want to see. Users should never get the impression that a field is filled if in reality it is not.

**Comment by Woseseltops** [comment]
Okay @susanodd completely rewrote how the list view works (with Ajax and everything, it's now much faster), so this was a nice opportunity to get used to this new approach.

**Comment by Woseseltops** [comment]
Happy @ocrasborn?

**Comment by susanodd** [comment]
I found the code that is causing this. It's hard-coded for display purposes, if there is no annotation translation to show, it shows English.
This would seem to have been introduced to avoid displaying nothing. (Context sensitive.)
I'll add a parameter so that the caller can dictate whether to show English or nothing.

**Comment by susanodd** [comment]
I fixed this. I'll push it as soon as I finish the other issue I'm working on, since I don't want to pull or merge yet.

**Comment by susanodd** [comment]
Oh, I see now that @Woseseltops also fixed this.
I solved it differently, I made a separate filter for the case when nothing should be displayed instead the default (it's hard-coded as 'eng' at the moment, that's what @Woseseltops commented out).
There are some places in the code where something has to be displayed, so the default language is needed.
However, we should use the default language from the settings, not English (as seems to be a problem in #579.) And either display nothing, or the default.
